libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1014,7 +1014,7 @@ bool TypeChecker::visit(TupleExpression const& _tuple)
 						fatalTypeError(components[i]->location(), "Invalid mobile type.");
 
 					if (i == 0)
-						inlineArrayType = types[i];
+						inlineArrayType = types[i]->mobileType();
 					else if (inlineArrayType)
 						inlineArrayType = Type::commonType(inlineArrayType, types[i]);
```

Comment:
`commonType(...)` will do that for us already. Furthermore, if we do that, we have to check for `types[i]->mobileType()` to be non-null (for `i=0` the check is in line 1013.

The source of this bug was that `->mobileType()` was not called for singletons because `Type::commonType()` was never invoked for them.